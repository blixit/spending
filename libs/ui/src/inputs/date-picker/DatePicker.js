import React from 'react';
import styled from 'styled-components';
import DatePicker, { registerLocale } from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import fr from 'date-fns/locale/fr'; // the locale you want

registerLocale('fr', fr); // register it with the name you want

/**
 * https://github.com/Hacker0x01/react-datepicker/issues/635
 *
 * export const popperPlacementPositions = [
  'auto',
  'auto-left',
  'auto-right',
  'bottom',
  'bottom-end',
  'bottom-start',
  'left',
  'left-end',
  'left-start',
  'right',
  'right-end',
  'right-start',
  'top',
  'top-end',
  'top-start'
]
 */
const StyledDatePicker = styled(DatePicker)`
  height: 30px;
  width: 80px;
  background: transparent;
  border: 0;
  border-bottom: 1px solid silver;
  line-height: 16px;
  font-size: 16px;
  &:focus{
    outline: none;
  }
`;

function DatePickerComponent (props) {
  const { onChange } = props;

  function change(date) {
    console.log({ date });

    // onChange(date.toLocaleDateString() + ' ' + date.toLocaleTimeString());
    onChange(date);
  };

  return (
    <StyledDatePicker
      dateFormat="dd/MM/yyyy"
      locale='fr'
      selected={props.selected}
      {...props}
      onChange={change}
    />
  );
}

export default DatePickerComponent;
