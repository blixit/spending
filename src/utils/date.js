/**
 * Formats a date for the backend
 * @param {Date} date the date to convert
 */
export const toBackendDate = (date) => {
  return date
    ? date.toLocaleDateString() + ' ' + date.toLocaleTimeString()
    : date;
};