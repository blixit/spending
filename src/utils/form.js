/**
* Removes the listed fields from the source
* @param {*} form The source object
* @param {*} fields The fields list
*/
export const clean = (form, fields) => {
  const result = {};
  // eslint-disable-next-line no-unused-vars
  for (const field in form) {
    if (!form.hasOwnProperty(field) || fields.includes(field)) {
      continue;
    }
    result[field] = form[field];
  }
  return result;
};
