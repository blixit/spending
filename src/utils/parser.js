export const stringify = data => {
  try {
    return JSON.stringify(data);
  } catch (error) {
    return null;
  };
}

export const parse = data => JSON.parse(data);