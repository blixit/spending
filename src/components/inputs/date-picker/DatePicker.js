import React from 'react';
import DatePicker from 'react-datepicker';

import 'react-datepicker/dist/react-datepicker.css';

function DatePickerComponent (props) {
  const { onChange } = props;

  return (
    <DatePicker selected={props.selected} onChange={onChange} {...props} />
  );
}

export default DatePickerComponent;
