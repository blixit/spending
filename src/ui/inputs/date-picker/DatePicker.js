import React, { useCallback } from 'react';
import styled from 'styled-components';
import DatePicker, { registerLocale } from 'react-datepicker';
import 'react-datepicker/dist/react-datepicker.css';
import fr from 'date-fns/locale/fr'; // the locale you want

import { locale } from 'utils/date';

registerLocale(locale, fr); // register it with the name you want

/**
 * https://github.com/Hacker0x01/react-datepicker/issues/635
 *
 * export const popperPlacementPositions = [
  'auto',
  'auto-left',
  'auto-right',
  'bottom',
  'bottom-end',
  'bottom-start',
  'left',
  'left-end',
  'left-start',
  'right',
  'right-end',
  'right-start',
  'top',
  'top-end',
  'top-start'
]
 */
const StyledDatePicker = styled(DatePicker)`
  height: 30px;
  width: 70px;
  background: transparent;
  border: 0;
  border-bottom: 1px solid silver;
  line-height: 16px;
  font-size: 0.8rem;
  color: ${({ theme }) => theme.primary()};

  &:focus{
    outline: none;
    border-bottom: 1px solid ${({ theme }) => theme.primary()};
  }
`;

const DatePickerComponent = props => {
  const { selected, onChange, arabic, placeholderText } = props;

  const change = useCallback(
    onChange ? date => onChange(date) : null,
    [onChange]
  );

  return (
    <StyledDatePicker
      dateFormat="dd/MM/yyyy"
      locale={locale}
      showPopperArrow={false}
      placeholderText={placeholderText || 'select a date'}
      popperPlacement={arabic ? 'bottom-end' : 'bottom-start'}
      selected={selected}
      {...props}
      onChange={change}
    />
  );
}

export default DatePickerComponent;
