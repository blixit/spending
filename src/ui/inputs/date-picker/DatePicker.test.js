import render from 'testing/render';

import { withTheme } from 'ui/theming/test.utils';

import DatePicker from './DatePicker';

describe('[inputs/date-picker] DatePicker', () => {
  it('renders properly', () => {
    const props = {
      onChange: jest.fn(),
      placeholderText: 'select the date',
      selected: Date.UTC(2019, 1, 17)
    };
    const tree = render(withTheme(DatePicker, props));
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
