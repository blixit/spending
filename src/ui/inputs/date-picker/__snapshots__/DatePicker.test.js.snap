// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[inputs/date-picker] DatePicker renders properly 1`] = `
<div
  className="react-datepicker-wrapper"
>
  <div
    className="react-datepicker__input-container"
  >
    <input
      className="sc-bdVaJa fEZIYW"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      placeholder="select the date"
      readOnly={false}
      type="text"
      value="17/02/2019"
    />
  </div>
</div>
`;
